docs/advanced-usage.rst

```diff
@@ -144,66 +182,146 @@ index ``cid``.
     :scale: 70%
     :align: center
 
+In order to prepare a transfer transaction, alice needs to provide at least
+three things:
+
+1. ``inputs`` -- one or more conditions that will be fulfilled.
+2. ``asset`` -- the asset being transferred.
+3. ``owners_after`` -- one or more public keys representing the new owner(s).
+
+To construct the input:
+
+.. code-block:: python
+
+    cid = 0
+    condition = tx['transaction']['conditions'][cid]
+    input_ = {
+        'fulfillment': condition['condition']['details'],
+        'input': {
+            'cid': cid,
+            'txid': tx['id'],
+        },
+        'owners_before': condition['owners_after'],
+    }
+
+The asset, can be directly retrieved from the input ``tx``:
```

Comment:
The current design of the driver is aiming to be a low-level layer, in which users are expected to have the know-how, or to go through a learning curve of learning how to construct the payloads on their own.

Once this low-level layer is solid enough we definitely wish to consider adding an additional layer that does provide an easy-to-use interface to compose crypto-conditions, and subsequently transactions.

I understand that the above may be debatable or questionable. I just created an issue which aims to address the underlying problem here, which is a need for clarification as to what is expected from the current driver.

The issue is at https://github.com/bigchaindb/bigchaindb-driver/issues/156

The current driver is the result of important design decisions, which unfortunately have yet to be documented, hence the above issue. It is hoped that once documented, multiple things will be clearer.